### PR TITLE
fix/GH-70 upcoming events page issue

### LINKED ISSUE #70: Upcoming events page doesn't always show next event
### Describe the bug

The upcoming events page doesn't show the next event when the following conditions are all true:
- next event is not published on meetup.
- The page is rendered before the 3rd Wednesday of the current month.

### System Info

```bash
not relevant
```

### Severity

major

### PR DESCRIPTION
**Pull request by FaberVitale**
### Description

Fixes #70 

### Other changes

- bumps node to 20.11.1.
- bumps pnpm to latest.
- updated readme.
- Use nvmrc to set node version in gh actions.
- Small overflow issues on mobile portrait.
- Added vitest as test runner (tests run on ci.yaml)

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -11,14 +11,15 @@ jobs:
...
+      TZ: 'Europe/Rome'
     steps:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
       - uses: pnpm/action-setup@v2
       - uses: actions/setup-node@v3
         with:
-          node-version: 'v18.17.0'
+          node-version-file: '.nvmrc'
```
Comment (gdorsi) [inline .github/workflows/ci.yaml:22]: Nice!

File: README.md [hunk 2]
```diff
@@ -26,7 +27,6 @@ The new RomaJS website built with [astro](https://astro.build/) and [SolidJS](ht
 - [x] github action pipeline that deploys to github pages
 - [x] blog post categories pages
 - [x] categories index page
-- [ ] blog post authors pages
```
Comment (gdorsi) [inline README.md:29]: :+1:

File: src/utils/meetup-events.ts [hunk 3]
```diff
@@ -27,3 +48,77 @@ export function computeMeetupEventsApiResponse(
...
+        !scheduledEvents.some(
+          (scheduled) =>
+            isSameDay(scheduled.epochTimeMs, placeholder.epochTimeMs) // filter out placeholder if overlaps with a scheduled event
+        )
+    );
+
+  return (scheduledEvents as UpcomingEvent[])
+    .concat(placeholderEvents)
+    .sort((a, b) => a.epochTimeMs - b.epochTimeMs);
+}
```
Comment (gdorsi) [inline src/utils/meetup-events.ts:124]: Looks like the logic is starting to be complex enough to justify some tests.

Do you have opinions about it?

If you agree I can start by creating an issue

File: src/components/EventsList/EventsList.tsx [hunk 4]
```diff
@@ -1,50 +1,53 @@
...
+                      {formatDate(lang, new Date(scheduled().dateTime))}
+                    </time>
+                    <h3 class={styles.eventHeading}>
+                      <a href={scheduled().eventUrl}>{scheduled().title}</a>
+                    </h3>
+                  </section>
+                </li>
+              )}
+            </Show>
+          </>
```
Comment (gdorsi) [inline src/components/EventsList/EventsList.tsx:50]: This component is becoming a bit hard to read.

How do you feel about splitting in multiple components?

**Review by gdorsi — APPROVED** [review]
Awesome job!



Thanks!

File: src/utils/meetup-events.ts (hunk 3 above)
Comment (FaberVitale) [reply src/utils/meetup-events.ts:124]: - Added vitest as test runner.
- Tests now run on CI
- added [meetup-events.test.ts](https://github.com/Roma-JS/roma-js-on-astro/blob/24a9b8207c5ca7d555be5bac8c354f9d47c128e8/src/utils/tests/meetup-events.test.ts) tests.

File: src/components/EventsList/EventsList.tsx (hunk 4 above)
Comment (FaberVitale) [reply src/components/EventsList/EventsList.tsx:37]: done in [feb1a5c](https://github.com/Roma-JS/roma-js-on-astro/pull/71/commits/feb1a5cb43c729605044f7f21757f7689dbf1c09)

File: src/utils/meetup-events.ts (hunk 3 above)
Comment (gdorsi) [reply src/utils/meetup-events.ts:124]: Awesome!

**Review by gdorsi — APPROVED** [review]

**Pull request merged**